niworkflows/anat/ants.py

```diff
@@ -855,6 +862,22 @@ def _argmax(in_dice):
             (apply_wm_prior, inu_n4_final, [("out", "weight_image")]),
         ])
         # fmt: on
+
+    if adaptive_bspline_grid:
+        # set INU bspline grid based on image shape
+        from ..utils.images import _bspline_grid
+        bspline_grid = pe.Node(niu.Function(function=_bspline_grid), name="bspline_grid")
+
+        # fmt:off
+        wf.connect([
+            (inputnode, bspline_grid, [(("in_files", _pop), "in_file")]),
+            (bspline_grid, inu_n4_final, [("out", "args")])
+        ])
+        # fmt:on
+    else:
+        # set INU bspline grid based on isotropic distance
+        inu_n4_final.inputs.bspline_fitting_distance = bspline_fitting_distance
```

Comment:
```suggestion
```

You can restore line 741 to pick `bspline_fitting_distance` as default, and the connection will override the value when `adaptive_bspline_grid` is true.